### DISCUSSION TITLE
Recognition Advice for my Dials

### DISCUSSION THREAD

**Discussion by saucedge** [jomjol/AI-on-the-edge-device#2205] [General]
### Issue

Thanks for an awesome project, I'm super excited to get it dialed in!  I am having a bit of trouble dialing in the robustness and I'm not sure whether I should continue pursuing image quality or if my analog dials and focal length are just too extreme to be reliably read.  For space constraints, I need the short focal length which is making the dials on the edge suffer from significant parallax.  And, there's a scuff on the meter cover that's proving to be really hard to mitigate no matter what I do to the lighting (dial 3 light spot).    The reflection on dial 2 can still be improved with additional baffling and things in my enclosure, I think.  Anyways, could I ask what you all would do to pursue robustness?  Is the parrallax acceptable?  Would you focus on lighting?  Image settings?  Would retraining the model help significantly?

![image](https://user-images.githubusercontent.com/52023198/225958847-a4ef20a0-7335-4b07-9406-fba408504d76.png)

The image shows only dial 3 wrong in this case, but all of the dials have had issues in various conditions.  

Thanks very much!

**Comment by caco3** [comment]
- The reference marks should be as far apart as possible to get the best alignment.
- the contrast is not that great, can you improve it?
- And reduce the reflection.
- The parallax should not be a problem.
- Check if your pointers are contained in https://jomjol.github.io/neural-network-analog-needle-readout/. If not, you should collect images so we can train the models better. See https://jomjol.github.io/AI-on-the-edge-device-docs/collect-new-images/

**Reply by saucedge** [reply]
Thanks for the advice!  

- For reference marks, that makes sense--do you think one of the zeros from '1,000,000' would work or is that too low of contrast?  
- For adding images to the training data, does the parrallax matter in this context?  I can pull images from the 1,000 dial for each number, but it will look significantly different than the 1,000,000 dial.
- I'll work to reduce the reflections and improve contrast in the meanwhile

Thanks again!

**Reply by caco3** [reply]
> For reference marks, that makes sense--do you think one of the zeros from '1,000,000' would work or is that too low of contrast?

Try it with the top left `B787`

> For adding images to the training data, does the parrallax matter in this context? I can pull images from the 1,000 dial for each number, but it will look significantly different than the 1,000,000 dial.

It actually does not look that much different! It should not be a problem.
